src/hooks/resources/device.ts

```diff
@@ -418,32 +422,24 @@ sbvrUtils.addPureHook('PATCH', 'resin', 'device', {
 				}),
 			);
 
-			// Also mark all image installs which are part of the current
-			// application as deleted.
-			waitPromises.push(
-				(args.request.custom.devices as Promise<AnyObject[]>).map(device => {
```

Comment:
This was actually an `AnyObject[]` so whilst the `.map` worked it did not wait on any of the returned promises